src/main/java/com/marklogic/client/ext/modulesloader/impl/BaseModulesFinder.java

```diff
@@ -64,18 +66,21 @@ protected void addAssetDirectories(Modules modules, String baseDir) {
 
 		// classpath needs the trailing / to find child dirs
 		findResources(baseDir, "*", "*/").stream().forEach(resource -> {
-			File f = null;
 			try {
-				f = resource.getFile();
-			} catch (IOException e) {}
-
-			if (!recognizedPaths.contains(resource.getFilename())) {
-				if (f == null || f.isDirectory()) {
-					if (dirs.indexOf(resource) < 0) {
+				File f = new File(resource.getURL().getFile());
```

Comment:
I recommend moving everything but the "dirs.add(resource)" line to a protected method - e.g. "isEligibleAssetDirectory" or "isValidAssetDirectory" (I think I prefer the former). Just gets all the logic into a small method that can be overridden if needed by a subclass.